spirv_common.hpp

```diff
@@ -296,6 +296,8 @@ struct SPIRType : IVariant
 		bool depth;
 		bool arrayed;
 		bool ms;
+		bool is_read = false;
```

Comment:
In MSL, textures and samplers are separate entities.textures can be declared with an `access`:

    enum class access { sample, read, write, read_write };

Default is `sample` for all but multisampled textures. Declaring a texture as `read_write` therefore, removes the `sample` declaration, which presumably does affect any performance optimizations around sampling.